### PR TITLE
 validateTileset asset check

### PR DESCRIPTION
**Pull request by Rudraksha20**
@lilleyse, 

worked on issue #91, check for a top-level asset property in Tileset. Also, check for property version inside asset to exist and have value '1.0'.

Included test cases for the same in validateTilesetSpec.js

### CODE REVIEW

File: validator/lib/validateTileset.js [hunk 1]
```diff
@@ -35,8 +35,21 @@ function validateTopLevel(tileset) {
     if (!defined(tileset.geometricError)) {
         return 'Tileset must declare its geometricError as a top-level property.';
     }
+
+    if(!defined(tileset.asset)){
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Style: For all these if statements change to `if (`

File: validator/lib/validateTileset.js [hunk 2]
```diff
@@ -35,8 +35,21 @@ function validateTopLevel(tileset) {
...
+        return 'Tileset must declare its asset as a top-level property.';
+    }
+
+    if(!defined(tileset.asset.version)){
+        return 'Tileset must declare its version property contained inside its top-level property asset.';
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Shorten the message to `Tileset must declare a version in its asset property`

File: validator/lib/validateTileset.js [hunk 3]
```diff
@@ -35,8 +35,21 @@ function validateTopLevel(tileset) {
...
+    }
+
+    if(tileset.asset.version !== '1.0'){
+        return 'Tileset must declare its asset.version property as always set to 1.0.';
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Shorten to `Tileset version must be 1.0`. Also include in the message the version it saw.

File: validator/lib/validateTileset.js [hunk 4]
```diff
@@ -35,8 +35,21 @@ function validateTopLevel(tileset) {
...
+    }
 }
 
+
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Remove empty line.

File: validator/specs/lib/validateTilesetSpec.js [hunk 5]
```diff
@@ -42,7 +42,9 @@ var sampleTileset = {
 
 describe('validateTileset', function() {
     it('succeeds for valid tileset', function(done) {
-        expect(validateTileset(sampleTileset)
+        var tileset = clone(sampleTileset, true);
+        tileset.asset.version = '1.0';
+        expect(validateTileset(tileset)
```
Comment (lilleyse) [inline validator/specs/lib/validateTilesetSpec.js]: Instead of fixing the version here can you set to 1.0 above? This was my mistake.

**Comment by Rudraksha20** [comment]
@lilleyse, made the recommended changes.

File: validator/lib/validateTileset.js [hunk 6]
```diff
@@ -35,6 +35,18 @@ function validateTopLevel(tileset) {
...
+    if (!defined(tileset.asset)){
+        return 'Tileset must declare its asset as a top-level property.';
+    }
+
+    if (!defined(tileset.asset.version)){
+        return 'Tileset must declare a version in its asset property';
+    }
+
+    if (tileset.asset.version !== '1.0'){
+        return 'Tileset version must be 1.0. ' + 'Tileset version provided: ' + tileset.asset.version;
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: The first two strings can be combined.

File: validator/lib/validateTileset.js [hunk 7]
```diff
@@ -35,6 +35,18 @@ function validateTopLevel(tileset) {
     if (!defined(tileset.geometricError)) {
         return 'Tileset must declare its geometricError as a top-level property.';
     }
+
+    if (!defined(tileset.asset)){
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Another style suggestion that I missed before, also add a space before the curly brace for these:

`if (!defined(tileset.asset)) {`

File: validator/lib/validateTileset.js (hunk 6 above)
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Same with the test.

**Comment by Rudraksha20** [comment]
@lilleyse updated.

**Comment by lilleyse** [comment]
Looks great, thanks @Rudraksha20!

**Pull request merged**